content/en/blog/_posts/2023-04-14-1.27-updates-on-speeding-up-pod-startup.md

```diff
@@ -0,0 +1,115 @@
+disabled by default (needs to be enabled explicitly). It aids pod startup by reducing unnecessary
+work during inactivity through replacing periodic polling. The current `Generic PLEG` incurs
+non-negligible overhead due to frequent container status polling, which worsens with Kubelet's
+parallelism, leading to poor performance and reliability issues. For these reasons, it is
+recommended that we use Evented PLEG instead. 
+
+Further details can be found in the KEP <https://kep.k8s.io/3386>.
+
+## Raise your pod resource limit if needed
+
+During start-up, some pods may consume a considerable amount of CPU or memory. If the CPU limit is
+low, this can significantly slow down the pod start-up process. To improve the memory management,
+Kubernetes v1.22 introduced a feature gate called MemoryQoS to kubelet. This feature enables
+kubelet to set memory QoS at container, pod, and QoS levels for better protection and guaranteed
+quality of memory when running with cgroups v2. Although it has benefits, it is possible that
+enabling this feature gate may affect the start-up speed of the pod if the pod startup consumes
+a large amount of memory.
+
+Kubelet configuration now includes `memoryThrottlingFactor`. This factor is multiplied by
+the memory limit or node allocatable memory to set the cgroupv2 memory.high value for enforcing
+MemoryQoS. Decreasing this factor sets a lower high limit for container cgroups, increasing reclaim
+pressure. Increasing this factor will put less reclaim pressure. The default value is 0.8 initially
+and will change to 0.9 in Kubernetes v1.27. This parameter adjustment can reduce the potential
+impact of this feature on pod startup speed.
+
+Further details can be found in the KEP <https://kep.k8s.io/2570>.
+
+## What's more?
+
+In Kubernetes v1.26, a new histogram metric `pod_start_sli_duration_seconds` was added for Pod
+startup latency SLI/SLO details. Additionally, the kubelet log will now display more information
+about pod start-related timestamps, as shown below:
+
+> Dec 30 15:33:13.375379 e2e-022435249c-674b9-minion-group-gdj4 kubelet[8362]: I1230 15:33:13.375359    8362 pod_startup_latency_tracker.go:102] "Observed pod startup duration" pod="kube-system/konnectivity-agent-gnc9k" podStartSLOduration=-9.223372029479458e+09 pod.CreationTimestamp="2022-12-30 15:33:06 +0000 UTC" firstStartedPulling="2022-12-30 15:33:09.258791695 +0000 UTC m=+13.029631711" lastFinishedPulling="0001-01-01 00:00:00 +0000 UTC" observedRunningTime="2022-12-30 15:33:13.375009262 +0000 UTC m=+17.145849275" watchObservedRunningTime="2022-12-30 15:33:13.375317944 +0000 UTC m=+17.146157970"
+
+The SELinux Relabeling with Mount Options feature moved to Beta in v1.27. This feature speeds up
+container startup by mounting volumes with the correct SELInux label instead of changing each file
+on the volumes recursively. Further details can be found in the KEP <https://kep.k8s.io/1710>.
```

Comment:
We should link to the blog article that https://github.com/kubernetes/website/pull/39836 adds (if it merges). The path will be `/blog/2023/04/18/kubernetes-1-27-efficient-selinux-relabeling-beta/`.